packages/backend/src/open_payments/auth/middleware.test.ts

```diff
@@ -85,47 +87,64 @@ describe('Auth Middleware', (): void => {
       })
       ctx.request.headers.authorization = ''
 
-      const throwSpy = jest.spyOn(ctx, 'throw')
       await expect(middleware(ctx, next)).resolves.toBeUndefined()
-      expect(throwSpy).toHaveBeenCalledWith(401, 'Unauthorized')
       expect(ctx.response.get('WWW-Authenticate')).toBe(
         `GNAP as_uri=${Config.authServerGrantUrl}`
       )
       expect(next).toHaveBeenCalled()
     })
 
-    test('throws error for unkonwn errors', async (): Promise<void> => {
+    test('throws error for unknown errors', async (): Promise<void> => {
       const middleware = createTokenIntrospectionMiddleware({
         requestType: type,
         requestAction: action,
         bypassError: true
       })
-      ctx.request.headers.authorization = ''
-      const error = new Error('Unknown')
-      ctx.throw = jest.fn().mockImplementation(() => {
-        throw error
-      }) as never
 
-      await expect(middleware(ctx, next)).rejects.toBe(error)
+      jest
+        .spyOn(tokenIntrospectionClient, 'introspect')
+        .mockResolvedValueOnce({ active: true, access: {} } as TokenInfo) // causes an error other than OpenPaymentsServerRouteError
+
+      expect.assertions(3)
```

Comment:
Handy `expect.assertions(num)` method when testing errors in `try/catch` to make sure the number assertions actually happened (instead of being ignored)